backend/dashboard/tests.py

```diff
@@ -1,3 +1,174 @@
+
+class TestCalculations(TestCase):
+    def setUp(self):
+        # FAKE THE DICTIONARIES
+        self.esg = {'3M Co': {'environment_level': 'High', 'social_grade': 'BB', 
+                              'governance_level': 'Medium', 'social_level': 'Medium', 
+                              'total_score': 1141, 'total_grade': 'BBB', 'social_score': 310, 
+                              'weburl': 'https://www.3m.com/', 'exchange': 'NEW YORK STOCK EXCHANGE, INC.', 
+                              'cik': 66740, 'logo': 'https://static.finnhub.io/logo/2a1802fa-80ec-11ea-a0f5-00000000092a.png', 
+                              'industry': 'Industrial Conglomerates', 'governance_score': 305, 
+                              'total_level': 'High', 'currency': 'USD', 'governance_grade': 'BB', 
+                              'environment_score': 526, 'last_processing_date': '16-04-2022', 
+                              'environment_grade': 'A', 'ticker': 'mmm'}, 
+                    'A O Smith Corp': {'environment_level': 'High', 'social_grade': 'BB',
+                                        'governance_level': 'Medium', 'social_level': 'Medium', 
+                                        'total_score': 1135, 'total_grade': 'BBB', 'social_score': 315, 
+                                        'weburl': 'https://www.aosmith.com/', 'exchange': 'NEW YORK STOCK EXCHANGE, INC.', 
+                                        'cik': 91142, 'logo': 'https://static.finnhub.io/logo/73381be8-80eb-11ea-b385-00000000092a.png', 
+                                        'industry': 'Building', 'governance_score': 310, 'total_level': 'High', 
+                                        'currency': 'USD', 'governance_grade': 'BB', 'environment_score': 510, 
+                                        'last_processing_date': '16-04-2022', 'environment_grade': 'A', 'ticker': 'aos'}}
+        
+        self.transactions = [{'action': 'declined', 'time_completed': '2024-08-31T06:02:27.687Z', 
+                                   'longitude': -113.807658, 'merchant_name': '3M Co', 'latitude': -42.372604, 
+                                   'customerID': 52, 'amount': 860.27, 'ip_address': '179.152.194.186'}, 
+                             {'action': 'declined', 'time_completed': '2023-12-07T08:07:20.451Z', 
+                                   'longitude': -1.121183, 'merchant_name': 'Smith Corp', 'latitude': 11.962175, 
+                                   'customerID': 52, 'amount': 144.53, 'ip_address': '173.64.65.25'}]
+    def test_get_closest_match(self):
+        # Test matching "3M Co" to "3M Co" in ESG_scores
+        result = _get_closest_match("3M", self.esg)
+        self.assertEqual(result, "3M Co")
+        
+        # Test matching "Smith Corp" to "A O Smith Corp"
+        result = _get_closest_match("Smith Corp", self.esg)
+        self.assertEqual(result, "A O Smith Corp")
+        
+        # Test when there's no match
+        result = _get_closest_match("Uber", self.esg)
```

Comment:
Uber doesn't have an ESG score in our ESG database, so i think the test makes sense

Reply:
In my opinion, we should not use Uber because to me it seems that Uber's ESG not being present is a dataset problem. With this test case, we want to cover the logic (a part of the user story) where the user purchases from a brand that is a local company and thus environmental impact can be considered to be 0. 

But I do understand, if we want to use another test to test more comprehensive user story, we can do that in another test? I just think this user story needs to be covered some how. 

What do you guys think?

Reply:
Ohh that does make sense. I agree!